### PR TITLE
OSOE-49: Update member order and remove section about code islands

### PR DESCRIPTION
**Pull request by sarahelsaig**
OSOE-49

### CODE REVIEW

File: Library/DevelopmentGuidelines/CodeStyling.md [hunk 1]
```diff
@@ -23,45 +23,30 @@ Prefix private variables with an underscore (_).
...
+- readonly fields
+- other fields
+- properties
+- constructors
+- instance methods
+- static methods
+- inner classes
 
-Have a standard ordering of members depending on their visibility and whether they're instance- or class-level, etc.
+Within this order fields and properties are sorted private, protected, public.
```
Comment (0liver) [inline Library/DevelopmentGuidelines/CodeStyling.md]:
```suggestion
Within this order, fields and properties are sorted private, protected, public.
```

File: Library/DevelopmentGuidelines/CodeStyling.md [hunk 2]
```diff
@@ -23,45 +23,30 @@ Prefix private variables with an underscore (_).
...
-        }
-    }
-
-If you have multiple types (e.g. an interface and a class) defined in the same file, similarly divide them with two line breaks.
+Have a standard ordering of members depending on their visibility and whether they're instance- or class-level, etc.
 
-----------
+Notice the order: 
+- constant fields
+- readonly fields
```
Comment (0liver) [inline Library/DevelopmentGuidelines/CodeStyling.md:30]:
```suggestion
- constant fields
- static fields
- readonly fields
```

File: Library/DevelopmentGuidelines/CodeStyling.md [hunk 3]
```diff
@@ -23,45 +23,30 @@ Prefix private variables with an underscore (_).
...
     public class MyClass
     {
+        // Constant fields come before everything else.
+        public const string MyConst = "const";
+	
+        // Readonly fields should appear before non-readonly fields
+        private readonly string _myReadonly = "readonly";
+	
         // Static fields first
         private static string _myStaticField = "field";
```
Comment (0liver) [inline Library/DevelopmentGuidelines/CodeStyling.md:52]: These two should be swapped.

File: Library/DevelopmentGuidelines/CodeStyling.md [hunk 4]
```diff
@@ -73,45 +58,37 @@ Have a standard ordering of members depending on their visibility and whether th
...
 
-
         // Private methods
         private void MyPrivateMethod()
         {
         }
 
-
         // Static methods
         private static void MyStaticMethod()
```
Comment (0liver) [inline Library/DevelopmentGuidelines/CodeStyling.md]: Let's add that `public static void MyPublicStaticMethod()` above this one, so we don't have to interpolate. (I read your comment above, but still.)

**Review by 0liver — CHANGES_REQUESTED** [review]

File: Library/DevelopmentGuidelines/CodeStyling.md (hunk 2 above)
Comment (sarahelsaig) [reply Library/DevelopmentGuidelines/CodeStyling.md:30]: That's not right. All readonly fields must come before non-readonly fields or the analyzer will throw a warning.

File: Library/DevelopmentGuidelines/CodeStyling.md (hunk 3 above)
Comment (sarahelsaig) [reply Library/DevelopmentGuidelines/CodeStyling.md:52]: No,  for the reason I wrote above.

File: Library/DevelopmentGuidelines/CodeStyling.md (hunk 4 above)
Comment (sarahelsaig) [reply Library/DevelopmentGuidelines/CodeStyling.md]: Ok added.

File: Library/DevelopmentGuidelines/CodeStyling.md (hunk 2 above)
Comment (0liver) [reply Library/DevelopmentGuidelines/CodeStyling.md:30]: I don't get any analyzer warning for this:

```c#
public class PrivacyConsentService : IPrivacyConsentService
{
    private const string _const = nameof(PrivacyConsentService);
    private static string _prop = typeof(PrivacyConsentService).FullName;

    private readonly UserManager<IUser> _userManager;
    private readonly IUserService _userService;

    // ...
}
```

Comment (0liver) [reply Library/DevelopmentGuidelines/CodeStyling.md:30]: It also makes the most sense to me to have `const`, `static`, and instance members in that order for fields, and in the reverse order for methods. And that's the order I've defined in my file layout, too:

image

Comment (sarahelsaig) [reply Library/DevelopmentGuidelines/CodeStyling.md:30]: This is what you get in OSOCE if you flip the two fields as you suggested.
image

Comment (sarahelsaig) [reply Library/DevelopmentGuidelines/CodeStyling.md:30]: See https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1214.md

Comment (0liver) [reply Library/DevelopmentGuidelines/CodeStyling.md:30]: OK, we have quite a few dimensions to map onto a single one (order) here:

- `const`, `static`, or `instance`
- `public`, `protected`, `private`
- `readonly`, not readonly

Let's follow the analyzer then.

Tbh, I haven't encountered any situation in the last year where a static field would be non-readonly.

File: Library/DevelopmentGuidelines/CodeStyling.md [hunk 5]
```diff
@@ -23,45 +23,30 @@ Prefix private variables with an underscore (_).
...
+- readonly fields
+- other fields
+- properties
+- constructors
+- instance methods
+- static methods
+- inner classes
 
-Have a standard ordering of members depending on their visibility and whether they're instance- or class-level, etc.
+Within this order, fields and properties are sorted private, protected, public.
```
Comment (0liver) [inline Library/DevelopmentGuidelines/CodeStyling.md]:
```suggestion
Within this order, fields and properties are sorted first by static vs. non-static, and inside each of these by private, protected, public.

```

**Review by 0liver — CHANGES_REQUESTED** [review]

**Comment by 0liver** [comment]
This one slipped through, but let's finish it.

File: Library/DevelopmentGuidelines/CodeStyling.md (hunk 5 above)
Comment (sarahelsaig) [reply Library/DevelopmentGuidelines/CodeStyling.md]: ok

**Review by 0liver — APPROVED** [review]

**Pull request merged**